fido/exceptions.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+
+class BaseTimeoutError(Exception):
+    """
+    Base class for all errors due to timeouts.
+    """
+
+
+class ConnectTimeoutError(BaseTimeoutError):
+    """
+    Connection took too long to establish
+    """
+
+
+class HTTPTimeoutError(BaseTimeoutError):
+    """
```

Comment:
mmh, it's not about hiding something secret and the mapping is not so straighforward.

**Mapping (more or less):**
HTTPTimeoutError -> twisted.web._newclient.ResponseNeverReceived
ConnectTimeoutError -> twisted.internet.error.ConnectError

**Twisted alternative**
I could just let all Twisted exceptions bubble up, but in that case if you are a Fido user, you are required to also understand Twisted, and open Twisted code to understand what exceptions are thrown in case of connection or timeout issues (the naming in this case is confusing, those would be twisted.web._newclient.ResponseNeverReceived and twisted.internet.error.ConnectError). With this Fido change, the work is done for you.

This abstraction is particularly useful to implement retries at the user-app level. If you want to just retry in case any connection issues or timeout occurred, you could just catch the Base class.

**Why this PR?**
I am actually open to implement any of the two solutions, but **a change in the exception handling and contract (major version bump) is unfortunately required** as in the current implementation there is no distinction between HTTP Timeouts or Connect Errors (they are grouped and hidden) and this is required to implement retries mechanisms with SmastStack.